### PR TITLE
feat: mock intl api

### PR DESCRIPTION
**Pull request by BenIsenstein**
#### Purpose (TL;DR) - Mock the global Intl API

Solve the breaking of timer-dependant methods on the Intl global object by including a mocked version

#### Background (Problem in detail)

Even after calling install(), the Intl.DateTimeFormat() class will currently use the real time of the system, not the time specified by the fake clock.

Affected methods: DateTimeFormat.format(), DateTimeFormat.formatToParts()

My repo demonstrating the problem, using Jest (which uses sinonjs/fake-timers): https://github.com/BenIsenstein/jest_fix_intl_fake_timers

The fix works by following the patterns used to mock the global Date class and mocking the Intl object. A producer function is written to return an Intl object largely identical to the original, except for the DateTimeFormat() class. DateTimeFormat has several methods that are time-sensitive, and depend on the system time when no argument is passed. These are intercepted and made to default to the clock time.

```javascript
//eslint-disable-next-line jsdoc/require-jsdoc
function createIntl() {
    const ClockIntl = { ...NativeIntl };

    ClockIntl.DateTimeFormat = function (...args) {
        const realFormatter = new NativeIntl.DateTimeFormat(...args);
        const formatter = {};

        ["formatRange", "formatRangeToParts", "resolvedOptions"].forEach(
            (method) => {
                formatter[method] =
                    realFormatter[method].bind(realFormatter);
            }
        );

        ["format", "formatToParts"].forEach((method) => {
            formatter[method] = function (date) {
                return realFormatter[method](date || ClockIntl.clock.now);
            };
        });

        return formatter;
    };

    ClockIntl.DateTimeFormat.prototype = Object.create(
        NativeIntl.DateTimeFormat.prototype
    );
    ClockIntl.DateTimeFormat.prototype.constructor =
        ClockIntl.DateTimeFormat;
    ClockIntl.DateTimeFormat.supportedLocalesOf =
        NativeIntl.DateTimeFormat.supportedLocalesOf;

    return ClockIntl;
}
```

This fake Intl object is provided to the Global object when install() is called.

Line 1176, inside createClock()
```javascript
if (intlPresent) {
    clock.Intl = createIntl();
    clock.Intl.clock = clock;
}
```

and the original Intl object is restored during uninstall().

Line 979 inside hijackMethod()
An else-if clause added to handle hijacking Intl. Because Intl isn't a straightforward global function, rather a global object used like a module, it can't be mocked using the default else clause at the end of hijackMethod(). It has its own clause to set the mocked Intl object globally. 
```javascript
else if (method === "Intl") {
    target[method] = clock[method];
```

I'm fairly confident on implementation, however less confident on the tests. All advice is appreciated!

Ben

### CODE REVIEW

**Comment by fatso83** [comment]
Love when wanted features come with a PR <3 I am off on holiday with just a small mobile screen unsuitable for reviewing code, but hopefully someone else could have a looksie?

**Comment by benjamingr** [comment]
I'm on medical leave I'll try to take a look next week and maybe ask Erick from Node to look

**Comment by ErickWendel** [comment]
As I don't have much context I think it LGTM

**Comment by BenIsenstein** [comment]
Hi all, it looks like the issue in the test suite is being caused by a difference between the mochify and node runtimes. `Intl.DateTimeFormat.resolvedOptions()` returns a different value between the two. Any suggestions on how to resolve this?

**Comment by fatso83** [comment]
It's not mochify (which is running Chrome headless) vs Node, but specifically Node 14 breaking. You see the tests passing in Node 16 and 18. 

If you look at [the engine comparisons page for Intl.DateTimeFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat#:~:text=Toggle%20history-,options.fractionalSecondDigits,-parameter) you can see that `fractionalSecondDigits` was only added in Node 14.6. Guessing we run a slightly older version. 

In any case, we only support Node LTS versions, and Node 14 LTS has reached its EOL, so we can just remove that :)

**Comment by fatso83** [comment]
I have now removed Node 14 from `main`. Does not block this and you do not _need_ to rebase to verify, but know you now 👍 

Do you deem it ready for final review and merge?

**Comment by BenIsenstein** [comment]
Rebased just to keep things clean (Y) yes, it's ready for final review!

File: CHANGELOG.md
Comment (fatso83) [inline CHANGELOG.md:5]: You can usually drop this. This is semi-auto-populated on `npm publish`.

File: src/fake-timers-src.js
Comment (fatso83) [inline src/fake-timers-src.js:212]: Drop this "tautology". The above stuff is only a custom hack for IE, which does not have Intl anyway. I will remove the above, as we no longer support IE.

File: src/fake-timers-src.js
Comment (fatso83) [inline src/fake-timers-src.js:537]: Could you add a check to verify that the `constructor` property is indeed the faked Intl object and not the native one? Otherwise we might as well drop it, as it's a behavior not specified by tests.

File: src/fake-timers-src.js
Comment (fatso83) [inline src/fake-timers-src.js:527]: From reading the code and checking against the PR description, this is basically the entire reason for this feature (right?): to get correct behaviour (formatting today's date) when leaving [the optional `date` argument out](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/formatToParts).

**Review by fatso83 — CHANGES_REQUESTED** [review]
This is an excellent PR. Really conforms to everything we want, so thank you very much! I'll merge it once the (tiny) changes have been done.

File: src/fake-timers-src.js
Comment (fatso83) [reply src/fake-timers-src.js:537]: Maybe also just add a basic tests for the supported locales?

File: src/fake-timers-src.js
Comment (BenIsenstein) [reply src/fake-timers-src.js:527]: Yes, that's exactly right! Expose the clock to the `DateTimeFormat` instance, pass the now value as the backup argument.

**Comment by BenIsenstein** [comment]
Thanks for the review! Changes:

- removed changelog entry
- removed IE tautology
- removed setting constructor (setting the prototype takes care of that already, good catch)
- added a test for the `supportedLocalesOf` method

**Review by fatso83 — APPROVED** [review]

**Pull request merged**
